extensions/eclipse-che-theia-plugin-remote/src/node/plugin-host-custom.ts

```diff
@@ -74,14 +75,19 @@ process.on('rejectionHandled', (promise: Promise<any>) => {
 });
 
 const emitter = new Emitter<string>();
-const rpc = new RPCProtocolImpl({
-  onMessage: emitter.event,
-  send: (m: string) => {
-    if (process.send) {
-      process.send(m);
-    }
+const rpc = new RPCProtocolImpl(
+  {
+    onMessage: emitter.event,
+    send: (m: string) => {
+      if (process.send) {
+        process.send(m);
+      }
+    },
   },
-});
+  {
```

Comment:
if not given couldn't we use default one in upstream (so no need to change che-theia ?, upstream theia having a default resolver on rpc protocolimpl )

Reply:
The default one _is_ used in upstream in all cases except on the plugin host process. Both changes here enable the plugin-host-specific reviver.

Reply:
We shouldn't have to import upstream reviver to give it to RPCProtocolImpl
if it requires a separate usecase then we should have like a new Impl class using the theia plug-in-ext reviewer

Reply:
It's not che-theia specific, it's process-specific. Since we're overriding the plugin process mechanism, we need to override the json marshalling. I've chosen to make the reviver/replacer a configuration object of the rpc protocol. That change has been approved (and now merged) upstream. This PR is just implementing the needed change to follow up in che-theia.

Reply:
process specific: Introduce stuff upstream without breaking downstream (for example provide new classes / etc that downstream products can use) and then complete the stuff (so downstream products are never broken)